### DISCUSSION TITLE
fallback:true not work when client-side page transitions

### DISCUSSION THREAD

**Discussion by sumiren** [vercel/next.js#40095] [Help]
I tried getStaticPaths with `fallback:true`, and noticed following behaviors.

* When reloading the page, Next.js shows the fallback page.
* When transiting to the page with next/link (client-side page transition), Next.js blocks transition and stays on previous page until the page is generated.

I wonder if the latter behavior is a spec.

**Comment by lucasbasquerotto** [comment]
When you use `fallback: true`, NextJS will return the cached page generated at build time in the 1st call, build it at the server side in background and return the new built page in the next requests, revalidating it when the time expires (in background). This can cause problems when using parameters/slugs in the path, because the 1st time they will be undefined.

You can use `fallback: 'blocking'` to build the page the first time, instead of returning the page generated at build time (so I advise to use `fallback: 'blocking'` instead of `fallback: true` in most cases). 

I'm not sure if this is your problem tough, but you can see if `fallback: 'blocking'` solves it.

**Reply by sumiren** [reply]
Thank you for your response!

I'll read your answer...

**Reply by sumiren** [reply]
I understood `fallback: "blocking'` behavior.
I'll stop using `fallback: "true'` if problem wasn't resolved.

Thanks!

**Comment by sumiren** [comment]
(I noticed that this problem is not specification conflicts between ISR and fallback, but it may be `fallback:true` behavior on client-side page transitions.)

**Comment by sumiren** [comment]
I rewrited contents.

**Comment by sumiren** [comment]
Maybe this discussion is similar to this
https://github.com/vercel/next.js/discussions/24265
